polyfills/contacts/contacts.js

```diff
@@ -0,0 +1,139 @@
+// mozContacts polyfill!
+// https://developer.mozilla.org/es/docs/Web/API/ContactManager
+
+// I have to define the service protocol, and this is as good place as any. The
+// protocol will be as follows:
+// * Each request will have a numeric request id, set by the client
+// * The answers will include the request id, so the client can know to which
+//   request the answer corresponds
+
+// For all the operations over a DeviceStorage object:
+//  * Request: { id: requestId,
+//               data: {
+//                 operation: methodName,
+//                 params: params
+//               }
+//             }
+// * Answer: { id: requestId,
+//             result|error: Whatever }
+
+// For the oncontactchange operation
+//  Request: { id: requestId,
+//             data: {
+//             operation: 'oncontactchange'
+//           },
+// Answer: Will be invoked when there's activity with
+// the MozContactChangeEvent (like) object:
+//    { id: requestId,
+//      event: MozContactChangeEvent }
+
+'use strict';
+
+/* globals FakeDOMRequest, FakeDOMCursorRequest, NavConnectHelper */
+
+(function(window) {
+
+  function debug(text) {
+    console.log('*-*-*- MozContacts PF: ' + text);
+  }
+
+  // Wishful thinking at the moment...
+  const CONTACTS_SERVICE = 'https://contactsservice.gaiamobile.org';
+
+  // It's nice being monothread...
+  var _currentRequestId = 1;
+
+  function _createAndQueueRequest(data, constructor) {
+    var request = new constructor(++_currentRequestId, data);
+    navConnPromise.
+          then(navConnHelper => navConnHelper.sendObject(request));
+    return request;
+  }
+
+  function _sendRequest(request) {
+    navConnPromise.
+          then(navConnHelper => navConnHelper.sendObject(commandObject));
+  }
+
+  window.navigator.mozContacts = {
+    _oncontactchange: null,
+    _oncontactchangeId: null,
+
+    get oncontactchange() {
+      return this._oncontactchange;
+    },
+
+    set oncontactchange(cb) {
+      this._oncontactchange = cb;
+      this._oncontactchangeId = this._oncontactchangeId || ++_currentRequestId;
```

Comment:
That way, once a handler is set we don't need to generate more traffic on the server (which will be mostly a NOP) only to change the local processAnswer